text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+While not a part of this RFC, end users can implement additional projects such as [in-toto](https://in-toto.io/) to build more advanced verification and attestation of software.
+
+## Goals
+
+This proposal will enhance cargo and crates.io through adoption of the Sigstore capabilities and workflows for supporting both use cases as outlined above. Specifically, 
+
+* Adopt Sigstore infrastructure to allow for the signing of crates for use both above.
+* Facilitate workflows in both Cargo and crates.io to allow for signature generation in the former case and verification in the latter case of crates and their dependencies by crate consumers (a note on dependencies: the depth level should be 1 for verification in this first phase)
+
+This implementation will support all three Sigstore implementation models which will be effectively transparent to both crates.io maintainers and crates.io consumers. At a high level the three Sigstore implementation models are:
+
+1. Use of the public Sigstore common good service, signing with the Sigstore public root.
+1. Signature operations via a private fulcio instance with intermediate signer issued internally, but signature metadata pushed to public Rekor instance.
+1. Full internal Sigstore infra used with internal consumption only software.
+
+# Non Goals
+
+* Implementing TUF - this proposal describes a sequenced implementation with TUF in a phase 2. 
+* Signing crates.io index - this is covered by other proposals, but is discussed in the context of TUF later in this document.
+* Extending or altering the Identity approach of Sigstore, i.e. OIDC based
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Crate signatures and verifications are a way to tie the authenticity and integrity of a crate based on public key cryptography. The implication is that, as an end user, you can be sure that the crates that you download from crates.io have not been tampered with, and that they are signed by the correct owner of the package.
+
+The process of `signing` involves generating a signature for a crate, using a private key, and attaching that signature when publishing a crate. The process of `verification` involves downloading a crate (using cargo) and using a _trusted_ public key to verify the authenticity (the signer) and integrity (the contents) of a crate.
+
+To simplify key management for users, cargo supports using [Sigstore](https://sigstore.dev) as a way to generate ephemeral keys based on an identity, and as an immutable log that can be monitored and audited in the event of a compromise. By default, a publicly available instance ensures that you do not need to manage this infrastructure yourself. If you are running your own crate registry, Sigstore instance and identify provider, you can configure cargo to use that.
```

Comment:
This is explaining a bit more than I'd consider reasonable for a guide-level explanation: for example, a user doesn't really need to know exactly what `cargo publish` does in order to use it effectively. Instead, this hints at auditing without explaining to users how they could do that, other than a vague notion of that being good.

I think that clarifying the guarantees of an immutable log would be better than simply talking about SigStore here.